.github/workflows/tests.yml

```
@@ -0,0 +1,252 @@
+        echo "Tests were skipped because:" >> $GITHUB_STEP_SUMMARY
+        echo "- No relevant code or test files were changed" >> $GITHUB_STEP_SUMMARY  
+        echo "- Last workflow run was successful" >> $GITHUB_STEP_SUMMARY
+        echo "" >> $GITHUB_STEP_SUMMARY
+        echo "To force test execution, add `[run tests]` to your commit message." >> $GITHUB_STEP_SUMMARY
+        
+  # Code quality checks - always run for PR changes
+  code-quality:
+    runs-on: ubuntu-latest
+    needs: changes
+    if: github.event_name == 'pull_request' && needs.changes.outputs.should-run-tests == 'true'
+    
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)